integration-test/src/test/java/org/ehcache/integration/ExpiryEventsTest.java

```diff
@@ -0,0 +1,98 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.integration;
+
+import org.ehcache.Cache;
+import org.ehcache.CacheManager;
+import org.ehcache.CacheManagerBuilder;
+import org.ehcache.config.CacheConfigurationBuilder;
+import org.ehcache.config.ResourcePoolsBuilder;
+import org.ehcache.config.units.EntryUnit;
+import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.event.CacheEvent;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.EventFiring;
+import org.ehcache.event.EventOrdering;
+import org.ehcache.event.EventType;
+import org.ehcache.expiry.Duration;
+import org.ehcache.expiry.Expirations;
+import org.junit.Test;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+/**
+ * Created by alsu on 06/08/15.
+ */
+public class ExpiryEventsTest {
+  @Test
+  public void testExpiredEvents() throws Exception {
+    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+        .withCache("tieredCache",
+            CacheConfigurationBuilder.newCacheConfigurationBuilder()
+                .withExpiry(Expirations.timeToLiveExpiration(new Duration(1, TimeUnit.SECONDS)))
+                .withResourcePools(ResourcePoolsBuilder.newResourcePoolsBuilder()
+                        .heap(3, EntryUnit.ENTRIES)
+                        .offheap(10, MemoryUnit.MB)
+                )
+                .buildConfig(Long.class, String.class))
+        .build(false);
+    cacheManager.init();
+
+    final List<Long> expiredKeys = new CopyOnWriteArrayList<Long>();
+
+    Cache<Long, String> tieredCache = cacheManager.getCache("tieredCache", Long.class, String.class);
+
+    tieredCache.getRuntimeConfiguration().registerCacheEventListener(new CacheEventListener<Long, String>() {
+      @Override
+      public void onEvent(CacheEvent<Long, String> event) {
+        expiredKeys.add(event.getKey());
+      }
+    }, EventOrdering.ORDERED, EventFiring.SYNCHRONOUS, EnumSet.of(EventType.EXPIRED));
+
+    tieredCache.put(1L, "one");
+    tieredCache.put(2L, "two");
+    tieredCache.put(3L, "three");
+    tieredCache.put(4L, "four");
+    tieredCache.put(5L, "five");
+
+    tieredCache.get(1L);
+    tieredCache.get(2L);
+    tieredCache.get(3L);
+
+    try {
+      Thread.sleep(1100);
```

Comment:
No. We can't have tests with Thread.sleep() in them. The time source is pluggable, so make use of that.